packages/headless/src/controllers/insight/category-facet/headless-insight-category-facet.ts

```diff
@@ -0,0 +1,233 @@
+import {
+  buildCoreCategoryFacet,
+  CategoryFacetSearch,
+  CategoryFacetValue,
+  CoreCategoryFacet,
+  CoreCategoryFacetState,
+} from '../../core/facets/category-facet/headless-core-category-facet';
+import {CategoryFacetOptions} from '../../core/facets/category-facet/headless-core-category-facet-options';
+import {CategoryFacetSortCriterion} from '../../../features/facets/category-facet-set/interfaces/request';
+import {InsightEngine} from '../../../insight.index';
+import {
+  logFacetClearAll,
+  logFacetDeselect,
+  logFacetSelect,
+  logFacetShowLess,
+  logFacetShowMore,
+  logFacetUpdateSort,
+} from '../../../features/facets/facet-set/facet-set-insight-analytics-actions';
+import {
+  executeSearch,
+  fetchFacetValues,
+} from '../../../features/insight-search/insight-search-actions';
+import {
+  categoryFacetSearchSet,
+  categoryFacetSet,
+  configuration,
+  search,
+} from '../../../app/reducers';
+import {
+  CategoryFacetSearchSection,
+  CategoryFacetSection,
+  ConfigurationSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+
+export interface InsightCategoryFacetOptions extends CategoryFacetOptions {}
+
+export interface InsightCategoryFacetProps {
+  /**
+   * The options for  `InsightCategoryFacet` controller
+   */
+  options: InsightCategoryFacetOptions;
+}
+
+export interface InsightCategoryFacet extends CoreCategoryFacet {
```

Comment:
I guess you could only specify `facetSearch` and `state` in the interface since you are extending `CoreCategoryFacet`.